src/content-type.zig

```diff
@@ -0,0 +1,171 @@
+base: ContentBase,
+parameter: ?CharSet = null,
+
+pub const default: ContentType = .{ .base = .{ .text = .html } };
+
+pub const ContentType = @This();
+
+pub const ContentBase = union(Base) {
+    application: Application,
+    audio: Audio,
+    font: Font,
+    image: Image,
+    text: Text,
+    video: Video,
+    /// Multipart types
+    multipart: MultiPart,
+    message: MultiPart,
+
+    pub fn string(comptime cb: ContentBase) [:0]const u8 {
+        return switch (cb) {
+            inline else => |tag| @tagName(cb) ++ "/" ++ @tagName(tag),
+        };
+    }
+};
+
+pub const Base = enum {
+    // Basic types
+    application,
+    audio,
+    font,
+    image,
+    text,
+    video,
+    /// Multipart types
+    multipart,
+    message,
+
+    pub fn isMultipart(b: Base) bool {
+        return switch (b) {
+            .multipart, .message => true,
+            else => false,
+        };
+    }
+};
+
+pub const Application = enum {
+    @"x-www-form-urlencoded",
+    @"x-git-upload-pack-request",
+    @"octet-stream",
+
+    pub fn string(comptime app: Application) [:0]const u8 {
+        return switch (app) {
+            inline else => |r| @typeName(@This())[13..] ++ "/" ++ @tagName(r),
+        };
+    }
+
+    test "Application.string" {
+        // This should be a lowercase A, but I don't know how much time to
+        // invest into this yet.
+        try std.testing.expectEqualStrings(
+            "Application/octet-stream",
+            Application.@"octet-stream".string(),
+        );
+    }
+};
+
+pub const Audio = enum {
+    ogg,
+};
+
+pub const Font = enum {
+    otf,
+    ttf,
+    woff,
+};
+
+pub const Image = enum {
+    png,
+    jpeg,
+};
+
+pub const Text = enum {
+    plain,
+    css,
+    html,
+    javascript,
+};
+
+pub const Video = enum {
+    mp4,
+};
+
+pub const MultiPart = enum {
+    mixed,
+    @"form-data",
+};
+
+pub const CharSet = enum {
+    @"utf-8",
+};
+
+fn a(comptime b: ContentBase) [:0]const u8 {
```

Comment:
It looks like this function isn't used anywhere.

Reply:
good catch, was for debugging, thank you!